File: csharp/leet_code/950.cs
public class Solution
{
    public int[] DeckRevealedIncreasing(int[] deck)
    {
        Array.Sort(deck);
        var n = deck.Length;
        var ans = new int[n];
        var q = new Queue<int>(Enumerable.Range(0, n));

        for (var i = 0; i < n; i++)
        {
            ans[q.Dequeue()] = deck[i];
            if (q.Count > 0)
            {
                q.Enqueue(q.Dequeue());
            }
        }

        return ans;
    }
}

File: php/leet_code/950.php
<?php

    class Solution
    {

        /**
         * @param Integer[] $deck
         * @return Integer[]
         */
        function deckRevealedIncreasing($deck)
        {
            $n = count($deck);
            sort($deck);
            $q = new SplQueue();
            $ans = array_fill(0, $n, 0);
            for ($i = 0; $i < $n; $i++) {
                $q->enqueue($i);
            }
            for ($i = 0; $i < $n; $i++) {
                $idx = $q->dequeue();
                $ans[$idx] = $deck[$i];
                if (!$q->isEmpty()) {
                    $q->enqueue($q->dequeue());
                }
            }
            return $ans;
        }
    }
